grpc/balancer.go

```diff
@@ -0,0 +1,112 @@
+package grpc
+
+import (
+	"sync"
+
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+)
+
+type nameRoundRobin struct {
+	names     chan []grpc.Address
```

Comment:
Add some comments describing what each of these synchronization primitives does (in particular the two chans)